packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.ts

```diff
@@ -78,17 +79,26 @@ export const useFluentProvider_unstable = (
     theme: mergedTheme,
     // eslint-disable-next-line @typescript-eslint/naming-convention
     overrides_unstable: mergedOverrides,
-    themeClassName: useFluentProviderThemeStyleTag({ theme: mergedTheme, targetDocument }),
+    themeClassName: styleTagId,
 
     components: {
       root: 'div',
+      serverStyle: 'style',
     },
 
     root: getNativeElementProps('div', {
       ...props,
       dir,
       ref: useMergedRefs(ref, useFocusVisible<HTMLDivElement>({ targetDocument })),
     }),
+
+    serverStyle: resolveShorthand(props.serverStyle, {
+      required: useIsInSSRContext(),
+      defaultProps: {
+        id: styleTagId,
```

Comment:
It should also take `nonce` attribute

Reply:
what should the default value of `nonce` be?

Reply:
Actually, it should take all attributes, i.e.:

```js
  // See `useFluentProviderThemeStyleTag`
  const renderer = useRenderer_unstable();
  const styleElementAttributes = renderer.styleElementAttributes;
```